syntax = "proto3";

package bacs.problem.single;
option go_package = "single";

import "bacs/problem/single/process/settings.proto";
import "bacs/problem/single/testing.proto";

message ProblemExtension {
  Tests tests = 1;
}

message Tests {
  repeated string data_set = 1;
  repeated string test_set = 2;
}

message ProfileExtension {
  repeated TestGroup test_group = 1;
}

message Dependency {
  enum Requirement {
    ALL_OK = 0;
    ALL_FAIL = 1;
    AT_LEAST_ONE_OK = 2;
    AT_MOST_ONE_FAIL = 3;
    AT_LEAST_HALF_OK = 4;
  }
  string test_group = 1;
  Requirement requirement = 2;  // default = ALL_OK
}

message TestGroup {
  string id = 1;
  single.process.Settings process = 2;
  TestSequence tests = 3;
  repeated Dependency dependency = 4;
  // partially per-test if tests.continue_condition == ALL
  // conditionally if tests.continue_condition == WHILE_OK
  int64 score = 5;
}
